apps/input-descriptor-to-credential/src/validators/no-additional-properties-allowed.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2021 - 2023 Energy Web Foundation
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {
+  registerDecorator,
+  ValidationOptions,
+  ValidatorConstraint,
+  ValidatorConstraintInterface
+} from 'class-validator';
+import { ValidationArguments } from 'class-validator/types/validation/ValidationArguments';
+
+@ValidatorConstraint({ async: false })
+export class NoAdditionalPropertiesConstraint implements ValidatorConstraintInterface {
+  validate(value: any): boolean {
+    return !JSON.stringify(value).match('"additionalProperties":true');
+  }
+
+  defaultMessage(validationArguments?: ValidationArguments): string {
+    return `${validationArguments.property} should have all additionalProperties properties set to false`;
```

Comment:
I would add a bit more description here on why this must be the case.
```suggestion
    return `${validationArguments.property} should have all additionalProperties properties set to false.
                 This is because JSON schema with additionalProperties set to true may be set translated into JSON which has additional unexpected properties.`;
```